source/type/pb_converter.h

```diff
@@ -58,17 +72,329 @@ namespace navitia {
         struct Thermometer;
     }
 }
-namespace pt = boost::posix_time;
 
-// we don't use a boolean to have a more type checks
-enum class DumpMessage {
-    Yes,
-    No
+using jp_pair = std::pair<const navitia::routing::JpIdx, const navitia::routing::JourneyPattern&>;
+using jpp_pair = std::pair<const navitia::routing::JppIdx, const navitia::routing::JourneyPatternPoint&>;
+
+namespace navitia {
+
+struct VjOrigDest{
+    const nt::VehicleJourney* vj;
+    const nt::StopPoint* orig;
+    const nt::StopPoint* dest;
+    VjOrigDest(const nt::VehicleJourney*vj, nt::StopPoint* orig, nt::StopPoint* dest):vj(vj),
+        orig(orig), dest(dest) {}
 };
 
-#define null_time_period boost::posix_time::time_period(boost::posix_time::not_a_date_time, boost::posix_time::seconds(0))
+struct VjStopTimes{
+    const nt::VehicleJourney* vj;
+    const nt::StopTime* orig;
+    const nt::StopTime* dest;
+    VjStopTimes(const nt::VehicleJourney* vj, const nt::StopTime* orig, const nt::StopTime* dest): vj(vj),
+        orig(orig), dest(dest){}
+};
+
+struct StopTimeCalandar{
+    const nt::StopTime* stop_time;
+    const navitia::DateTime& date_time;
+    boost::optional<const std::string> calendar_id;
+    StopTimeCalandar(const nt::StopTime* stop_time, const navitia::DateTime& date_time,
+                     boost::optional<const std::string> calendar_id):
+        stop_time(stop_time), date_time(date_time), calendar_id(calendar_id){}
+};
+
+struct WayCoord{
+    const ng::Way* way;
+    const nt::GeographicalCoord& coord;
+    const int number;
+    WayCoord(const ng::Way* way, const nt::GeographicalCoord& coord,
+             const int number):
+        way(way), coord(coord), number(number){}
+
+};
+
+/**
+ * get_label() is a function that returns:
+ * * the label (or a function get_label()) if the object has it
+ * * else return the name of the object
+ *
+ * Note: the trick is that int is a better match for 0 than
+ * long and template resolution takes the best match
+ */
+namespace detail {
+template<typename T>
+auto get_label_if_exists(const T* v, int) -> decltype(v->label) {
+    return v->label;
+}
+
+template<typename T>
+auto get_label_if_exists(const T* v, int) -> decltype(v->get_label()) {
+    return v->get_label();
+}
+
+template<typename T>
+auto get_label_if_exists(const T* v, long) -> decltype(v->name) {
+    return v->name;
+}
+}
+
+template<typename T>
+std::string get_label(const T* v) {
+    return detail::get_label_if_exists(v, 0);
+}
+inline pbnavitia::NavitiaType get_embedded_type(const nt::Calendar*) { return pbnavitia::CALENDAR; }
```

Comment:
IMHO, a more proper way to do it will be like what's done at 188

    template<typename T>
    pbnavitia::NavitiaType get_label {  return pbnavitia::UNKNOWN; /* ? */}

    template<>
    pbnavitia::NavitiaType<nt::Calendar> get_label {  return pbnavitia::CALENDAR; }

When you call:

     nt::Calendar* c = ... ;
     auto pb_type = get_embedded_type<std::decay<declype(c)>::type>();
     // Or
     // auto pb_type = get_embedded_type<nt::Calendar>();

The advantage is that you don't need to have a pointer to get the pb_type. 

In this PR, I'm ok with it